TrustedXForwarderProxiesUnknownHostnameFailureTest failing on Windows

We have seen multiple failures of `TrustedXForwarderProxiesUnknownHostnameFailureTest` lately.

The log is the following:

```
2023-10-23T19:15:44.3488972Z [INFO] Running io.quarkus.vertx.http.proxy.TrustedXForwarderProxiesUnknownHostnameFailureTest
2023-10-23T19:15:46.0525436Z 2023-10-23 19:15:45,766 INFO  [io.quarkus] (main) quarkus-vertx-http-deployment 999-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 0.584s. Listening on: http://localhost:8081
2023-10-23T19:15:46.4971597Z 2023-10-23 19:15:45,766 INFO  [io.quarkus] (main) Profile test activated. 
2023-10-23T19:15:46.4979008Z 2023-10-23 19:15:45,766 INFO  [io.quarkus] (main) Installed features: [cdi, security, smallrye-context-propagation, vertx]
2023-10-23T19:15:47.1086455Z 2023-10-23 19:15:47,076 WARN  [io.net.res.dns.DefaultDnsServerAddressStreamProvider] (vert.x-eventloop-thread-0) Default DNS servers: [/8.8.8.8:53, /8.8.4.4:53] (Google Public DNS as a fallback)
2023-10-23T19:15:47.2104672Z 2023-10-23 19:15:47,115 ERROR [io.qua.ver.cor.run.VertxCoreRecorder] (vert.x-eventloop-thread-0) Uncaught exception received by Vert.x: java.lang.NullPointerException: Cannot invoke "String.indexOf(int)" because "address" is null
2023-10-23T19:15:47.2107640Z 	at org.wildfly.common.net.Inet.parseInetAddress(Inet.java:629)
2023-10-23T19:15:47.2108853Z 	at org.wildfly.common.net.Inet.parseInetAddress(Inet.java:616)
2023-10-23T19:15:47.2110282Z 	at io.quarkus.vertx.http.runtime.ForwardedProxyHandler$1.handle(ForwardedProxyHandler.java:83)
2023-10-23T19:15:47.2112037Z 	at io.quarkus.vertx.http.runtime.ForwardedProxyHandler$1.handle(ForwardedProxyHandler.java:79)
2023-10-23T19:15:47.2113579Z 	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
2023-10-23T19:15:47.2114869Z 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2023-10-23T19:15:47.2116234Z 	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2023-10-23T19:15:47.2117548Z 	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
2023-10-23T19:15:47.2118880Z 	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
2023-10-23T19:15:47.2120244Z 	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)
2023-10-23T19:15:47.2121624Z 	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:184)
2023-10-23T19:15:47.2122994Z 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:51)
2023-10-23T19:15:47.2124352Z 	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2023-10-23T19:15:47.2125687Z 	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
2023-10-23T19:15:47.2127314Z 	at io.vertx.core.Promise.complete(Promise.java:66)
2023-10-23T19:15:47.2128566Z 	at io.vertx.core.impl.future.PromiseImpl.operationComplete(PromiseImpl.java:65)
2023-10-23T19:15:47.2130134Z 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
2023-10-23T19:15:47.2131712Z 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
2023-10-23T19:15:47.2133402Z 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
2023-10-23T19:15:47.2136060Z 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
2023-10-23T19:15:47.2137566Z 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
2023-10-23T19:15:47.2138998Z 	at io.netty.util.concurrent.DefaultPromise.setSuccess(DefaultPromise.java:97)
2023-10-23T19:15:47.2140425Z 	at io.vertx.core.dns.impl.DnsClientImpl$Query.handle(DnsClientImpl.java:344)
2023-10-23T19:15:47.2141816Z 	at io.vertx.core.dns.impl.DnsClientImpl$1.channelRead0(DnsClientImpl.java:89)
2023-10-23T19:15:47.2143207Z 	at io.vertx.core.dns.impl.DnsClientImpl$1.channelRead0(DnsClientImpl.java:83)
2023-10-23T19:15:47.2144901Z 	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
2023-10-23T19:15:47.2146903Z 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2023-10-23T19:15:47.2148990Z 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2023-10-23T19:15:47.2151042Z 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2023-10-23T19:15:47.2152911Z 	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
2023-10-23T19:15:47.2154855Z 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2023-10-23T19:15:47.2156947Z 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2023-10-23T19:15:47.2159238Z 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2023-10-23T19:15:47.2161284Z 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
2023-10-23T19:15:47.2163468Z 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
2023-10-23T19:15:47.2165755Z 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2023-10-23T19:15:47.2167742Z 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
2023-10-23T19:15:47.2169611Z 	at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:97)
2023-10-23T19:15:47.2171392Z 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
2023-10-23T19:15:47.2173064Z 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
2023-10-23T19:15:47.2174752Z 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
2023-10-23T19:15:47.2176121Z 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
2023-10-23T19:15:47.2177675Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
2023-10-23T19:15:47.2179318Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2023-10-23T19:15:47.2181037Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2023-10-23T19:15:47.2182398Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2023-10-23T19:15:47.2183095Z 
2023-10-23T19:16:17.1813867Z 2023-10-23 19:16:17,136 INFO  [io.quarkus] (main) quarkus-vertx-http-deployment stopped in 0.018s
2023-10-23T19:16:17.3778724Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 32.93 s <<< FAILURE! -- in io.quarkus.vertx.http.proxy.TrustedXForwarderProxiesUnknownHostnameFailureTest
2023-10-23T19:16:17.3787803Z [ERROR] io.quarkus.vertx.http.proxy.TrustedXForwarderProxiesUnknownHostnameFailureTest.testHeadersAreIgnored -- Time elapsed: 31.34 s <<< ERROR!
2023-10-23T19:16:17.3789554Z java.net.SocketTimeoutException: Read timed out
2023-10-23T19:16:17.3790613Z 	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:288)
2023-10-23T19:16:17.3792012Z 	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:314)
2023-10-23T19:16:17.3793324Z 	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:355)
2023-10-23T19:16:17.3794418Z 	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:808)
2023-10-23T19:16:17.3795517Z 	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:966)
2023-10-23T19:16:17.3796938Z 	at org.apache.http.impl.io.AbstractSessionInputBuffer.fillBuffer(AbstractSessionInputBuffer.java:161)
2023-10-23T19:16:17.3798485Z 	at org.apache.http.impl.io.SocketInputBuffer.fillBuffer(SocketInputBuffer.java:82)
2023-10-23T19:16:17.3800048Z 	at org.apache.http.impl.io.AbstractSessionInputBuffer.readLine(AbstractSessionInputBuffer.java:276)
2023-10-23T19:16:17.3801862Z 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
2023-10-23T19:16:17.3803588Z 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
2023-10-23T19:16:17.3805139Z 	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
2023-10-23T19:16:17.3806879Z 	at org.apache.http.impl.AbstractHttpClientConnection.receiveResponseHeader(AbstractHttpClientConnection.java:294)
2023-10-23T19:16:17.3808813Z 	at org.apache.http.impl.conn.DefaultClientConnection.receiveResponseHeader(DefaultClientConnection.java:257)
2023-10-23T19:16:17.3810780Z 	at org.apache.http.impl.conn.ManagedClientConnectionImpl.receiveResponseHeader(ManagedClientConnectionImpl.java:207)
2023-10-23T19:16:17.3812584Z 	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
2023-10-23T19:16:17.3814192Z 	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
2023-10-23T19:16:17.3815750Z 	at org.apache.http.impl.client.DefaultRequestDirector.tryExecute(DefaultRequestDirector.java:679)
2023-10-23T19:16:17.3817379Z 	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:481)
2023-10-23T19:16:17.3818998Z 	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
2023-10-23T19:16:17.3820488Z 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
2023-10-23T19:16:17.3822015Z 	at io.restassured.internal.RestAssuredHttpBuilder.doRequest(RestAssuredHttpBuilder.java:141)
2023-10-23T19:16:17.3823570Z 	at io.restassured.internal.http.HTTPBuilder.doRequest(HTTPBuilder.java:496)
2023-10-23T19:16:17.3824801Z 	at io.restassured.internal.http.HTTPBuilder.request(HTTPBuilder.java:453)
2023-10-23T19:16:17.3826023Z 	at io.restassured.internal.http.HTTPBuilder$request$2.call(Unknown Source)
2023-10-23T19:16:17.3827382Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
2023-10-23T19:16:17.3828884Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
2023-10-23T19:16:17.3830380Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:166)
2023-10-23T19:16:17.3832042Z 	at io.restassured.internal.RequestSpecificationImpl.sendHttpRequest(RequestSpecificationImpl.groovy:1480)
2023-10-23T19:16:17.3833833Z 	at io.restassured.internal.RequestSpecificationImpl.this$2$sendHttpRequest(RequestSpecificationImpl.groovy)
2023-10-23T19:16:17.3835395Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-10-23T19:16:17.3836889Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-10-23T19:16:17.3838730Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-10-23T19:16:17.3840154Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
```

@cescoffier any idea of what could go wrong?

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedProxyHandler.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedProxyHandler.java
@@ -79,7 +79,7 @@ public class ForwardedProxyHandler implements Handler<HttpServerRequest> {
                     new Handler<AsyncResult<String>>() {
                         @Override
                         public void handle(AsyncResult<String> stringAsyncResult) {
-                            if (stringAsyncResult.succeeded()) {
+                            if (stringAsyncResult.succeeded() && stringAsyncResult.result() != null) {
                                 var trustedIP = Inet.parseInetAddress(stringAsyncResult.result());
                                 if (trustedIP != null) {
                                     // create proxy check for resolved IP and proceed with the lookup

